src/Extension/Publishable/PublishableSiteTree.php

```diff
@@ -90,6 +91,14 @@ public function objectsToDelete($context)
      */
     public function urlsToCache()
     {
-        return [Director::absoluteURL($this->getOwner()->Link()) => 0];
+        $page = $this->getOwner();
+        if ($page instanceof RedirectorPage) {
```

Comment:
I still feel like explicitly calling out `RedirectorPage` is a bit of an anti-pattern here - unless other redirects are _already_ working correctly, and this problem really is specifically only affecting this page type.

Reply:
I think it's fine, RedirectorPage lives in the cms module, and cms is a requirement in composer.json

The reason this fix works is that staticpublish queue will:
- collect a list of **urls** to cache
- make a fake to those urls
- cache the results - if the returned http status code is 300-399, cache as a redirect, else cache as a regular page

If we use RedirectorPage::Link() - it'll return the url of the target page
However, if we use RedirectorPage::regularLink() we get the url of the RedirectorPage - which will return a 301/302 when you normally visit it

Reply:
My issue isn't that `RedirectorPage` might somehow be missing, but rather that it might not be the _only_ way for redirects to be present in a project. It looks like this module is intended to cache _any_ redirects, not just redirects from this page type.

So like I said: If other redirects _do_ still get cached correctly and this issue _is unique_ to the redirector page type, this solution is fine.
But if other redirects _are still not cached correctly_ then I think the argument @dhensby made on the original PR still stands.

Reply:
Have confirmed, existing redirect function correct.  It's only RedirectoPage that has the issue.

**MyRedirectPage.php**
```php
<?php

class MyRedirectPage extends Page
{
}
```

**MyRedirectPageController.php**
```php
<?php

use SilverStripe\CMS\Model\SiteTree;

class MyRedirectPageController extends PageController
{
    public function init()
    {
        parent::init();
        $this->redirect(SiteTree::get()->filter(['Title' => 'Contact us'])->first()->Link());
    }
}
```

Creating an instance of this page, publishing it and then running the queued job will will result in a static redirect files created in /public/cache/new-my-redirect-page.html + /public/cache/new-my-redirect-page.php which both contain redirects to `<doman>/contact-us/`

Reply:
>If we use RedirectorPage::Link() - it'll return the url of the target page

Bummer - that's the worst possible behavior for this module. If the `Link()` function is lying about the page's link then we can't do anything except what's being suggested, I guess.